core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -488,6 +488,10 @@ private synchronized void recordLeafSplitsSize()
         this.lastLeafSplitsSize = allSplits.size() - intermediateSplits.size();
     }
 
+    /**
+     * A class acting as a worker as part of threadpool that keep pulling pending splits,
```

Comment:
I think since it implements `Runnable`, the first part is kind of clear. May be just say something like below or remove

```
Keeps processing split one-by-one from the list of waiting splits.
```

Reply:
removed.